osquery/tables/networking/linux/process_open_sockets.cpp

```diff
@@ -8,179 +8,119 @@
  *  You may select, at your option, one of the above-listed licenses.
  */
 
-#include <arpa/inet.h>
-
 #include <osquery/core.h>
 #include <osquery/filesystem.h>
 #include <osquery/tables.h>
 
 #include "osquery/core/conversions.h"
+#include "osquery/filesystem/linux/proc.h"
 
 namespace osquery {
 namespace tables {
-
-// Linux proc protocol define to net stats file name.
-const std::map<int, std::string> kLinuxProtocolNames = {
-    {IPPROTO_ICMP, "icmp"},
-    {IPPROTO_TCP, "tcp"},
-    {IPPROTO_UDP, "udp"},
-    {IPPROTO_UDPLITE, "udplite"},
-    {IPPROTO_RAW, "raw"},
-};
-
-// A map of socket handles (inodes) to their pid and file descriptor.
-typedef std::map<std::string, std::pair<std::string, std::string> > InodeMap;
-
-std::string addressFromHex(const std::string &encoded_address, int family) {
-  char addr_buffer[INET6_ADDRSTRLEN] = {0};
-  if (family == AF_INET) {
-    struct in_addr decoded;
-    if (encoded_address.length() == 8) {
-      sscanf(encoded_address.c_str(), "%X", &(decoded.s_addr));
-      inet_ntop(AF_INET, &decoded, addr_buffer, INET_ADDRSTRLEN);
-    }
-  } else if (family == AF_INET6) {
-    struct in6_addr decoded;
-    if (encoded_address.length() == 32) {
-      sscanf(encoded_address.c_str(),
-             "%8x%8x%8x%8x",
-             (unsigned int *)&(decoded.s6_addr[0]),
-             (unsigned int *)&(decoded.s6_addr[4]),
-             (unsigned int *)&(decoded.s6_addr[8]),
-             (unsigned int *)&(decoded.s6_addr[12]));
-      inet_ntop(AF_INET6, &decoded, addr_buffer, INET6_ADDRSTRLEN);
-    }
-  }
-
-  return std::string(addr_buffer);
-}
-
-unsigned short portFromHex(const std::string &encoded_port) {
-  unsigned short decoded = 0;
-  if (encoded_port.length() == 4) {
-    sscanf(encoded_port.c_str(), "%hX", &decoded);
-  }
-  return decoded;
-}
-
-void genSocketsFromProc(const InodeMap &inodes,
-                        int protocol,
-                        int family,
-                        QueryData &results) {
-  std::string path = "/proc/net/";
-  if (family == AF_UNIX) {
-    path += "unix";
+QueryData genOpenSockets(QueryContext& context) {
+  // If a pid is given then set that as the only item in processes.
+  std::set<std::string> pids;
+  if (context.constraints["pid"].exists(EQUALS)) {
+    pids = context.constraints["pid"].getAll(EQUALS);
```

Comment:
There is no way to perform that check, because we are not locking the /proc folder; it is better to just attempt to enumerate them, and handle the error

Reply:
My idea was just to check that `pid` is a valid number and not a random character string. But indeed if it isn't a number then we'll handle the error the same way we handle passing a `pid` that doesn't exist.